postgres_composite_types/fields.py

```diff
@@ -1,36 +1,59 @@
 import json
+from typing import TYPE_CHECKING, Type
 
 from django.core.exceptions import ValidationError
 from django.db.backends.postgresql.base import (
     DatabaseWrapper as PostgresDatabaseWrapper,
 )
 from django.db.models import Field
 
+if TYPE_CHECKING:
+    from .composite_type import CompositeType
+
+
 __all__ = ["BaseField"]
 
 
+class DummyField(Field):
+    """
+    A dummy field added on every CompositeType, that behaves as the
+    type's primary key. This is a hack due to Django's requirement for
+    all models to have a primary key.
+    """
+
+    name = "_id_not_used"
+
+
 class BaseField(Field):
     """Base class for the field that relates to this type."""
 
-    Meta = None
+    _composite_type_model: Type["CompositeType"]
 
     default_error_messages = {
         "bad_json": "to_python() received a string that was not valid JSON",
     }
 
+    def deconstruct(self):
+        name, path, args, kwargs = super().deconstruct()
+
+        name = path.replace("postgres_composite_types.composite_type.", "")
```

Comment:
This feels a bit fragile. Also the shadowing makes it hard to appreciate what will happen.

Reply:
I agree. I am not sure how else to do it though. Postgres' migration generation system is poor in terms of customizability.